created new endpoint for getting habits by different parameters

LiliaMokhnatska opened the pull request:
## Summary Of Issue :
Create new controller to get habits by selected filters and also sort it



**Issue Link**

https://github.com/ita-social-projects/GreenCity/issues/5679



## Summary Of Changes :
1) created get  "/habit/search" in HabitController
2) added "/habit/search" in SecurityConfig
3) added tests in HabitControllerTest
4) added method getAllByDifferentParameters in HabitService
5) implemented method getAllByDifferentParameters in HabitServiceImpl 
6) added new queries  in HabitTranslationRepo.
7) changed ModelUtils
8) added  new tests to HabitServiceImplTest
9) added new method in CustomExceptionHandler



## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [ ] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

LiliaMokhnatska committed:
created new endpoint for getting habits by different parameters

LiliaMokhnatska committed:
added tests for more cases in HabitControllerTest

LiliaMokhnatska committed:
added tests for more cases in HabitControllerTest

LiliaMokhnatska committed:
renamed method in CustomExceptionHandler

LiliaMokhnatska committed:
changed CustomExceptionHandler, HabitController, HabitControllerTest

LiliaMokhnatska committed:
added test for one more case in HabitControllerTest

RomaMocherniuk commented on core/src/main/java/greencity/controller/HabitController.java:
```diff
@@ -138,6 +142,39 @@ public ResponseEntity<PageableDto<HabitDto>> getAllByTagsAndLanguageCode(
+        @RequestParam(required = false, name = "isCustomHabit") Optional<Boolean> isCustomHabit,
+        @RequestParam(required = false, name = "complexity") Optional<Integer> complexity,
+        @ApiIgnore Pageable pageable) throws BadRequestException {
+        if ((tags.isPresent() && !tags.get().isEmpty()) || isCustomHabit.isPresent() || complexity.isPresent()) {
```
please move it to method `isValid()`

LiliaMokhnatska replied:
changed

LiliaMokhnatska committed:
added isValid method in HabitController
changed method getAllByDifferentParameters in HabitServiceImpl
changed parameters in getAllByDifferentParametersArguments in HabitServiceImplTest

LiliaMokhnatska committed:
fixed code smell

LiliaMokhnatska committed:
changed getAllByDifferentParameters in HabitServiceImplTest

LiliaMokhnatska committed:
created private method createPageRequest() in HabitControllerTest
and changed SQL querys into HQL querys
